### PR TITLE
Adding Content-Length header only when one is not found during proxy

### PR DESCRIPTION
**Pull request by matt-wormley**
Don't clobber the HEAD content-length with zero

### CODE REVIEW

File: localstack/services/generic_proxy.py [hunk 1]
```diff
@@ -147,9 +147,10 @@ def forward(self, method):
             # copy headers and return response
             self.send_response(response.status_code)
             for header_key, header_value in iteritems(response.headers):
-                if header_key.lower() != 'Content-Length'.lower():
+                if header_key.lower() != 'Content-Length'.lower() or method == 'HEAD':
                     self.send_header(header_key, header_value)
-            self.send_header('Content-Length', '%s' % len(response.content))
+            if method != 'HEAD':
```
Comment (whummer) [inline localstack/services/generic_proxy.py]: In the (unlikely?) case that the `response` of a `HEAD` request does *not* contain a `Content-Length`, we'll end up sending no `Content-Length` header at all in the final response. Not sure if this is the desired behavior.

To be on the safe side, we could change it to something like this:

```
content_length_sent = False
for header_key, header_value in iteritems(response.headers):
    is_content_length_header = header_key.lower() != 'Content-Length'.lower()
    if is_content_length_header or method == 'HEAD':
        self.send_header(header_key, header_value)
        content_length_sent = content_length_sent or is_content_length_header
if not content_length_sent:
    self.send_header('Content-Length', '%s' % len(response.content))
```

Comment (matt-wormley) [inline localstack/services/generic_proxy.py]: good point. I'll make a change.

Comment (matt-wormley) [inline localstack/services/generic_proxy.py]: how about this? This generalizes the cases. Any content-length header is passed through and if none is found len(response.content) is added.

```
content_length_sent = False
for header_key, header_value in iteritems(response.headers):
    self.send_header(header_key, header_value)
    content_length_sent = content_length_sent or header_key.lower() == 'content-length'
if not content_length_sent:
    self.send_header('Content-Length', '%s' % len(response.content))
```

Comment (whummer) [inline localstack/services/generic_proxy.py]: Yes, that's a clean and nice solution. LGTM

**Comment by matt-wormley** [comment]
@whummer I'm not sure which check failed

**Comment by whummer** [comment]
The build error was a misconfiguration in the CI, which is now fixed. We still have a minor merge conflict due to another change that happened in parallel. Can you please fix the conflict (and ideally squash the 3 commits into 1), then we should be good to merge. Thanks again!

**Comment by whummer** [comment]
Thanks for updating the PR, merging now.

**Comment by matt-wormley** [comment]
good stuff! great to contribute!

**Pull request merged**